pkg/kube/container_oci.go

```diff
@@ -268,33 +266,50 @@ func (t *containerTranslator) configureProcess() error {
 		t.g.AddProcessAdditionalGid(uint32(gid))
 	}
 
-	aaProfile := security.GetApparmorProfile()
-	selinuxOptions := security.GetSelinuxOptions()
-
-	if aaProfile != "" && selinuxOptions != nil {
-		return fmt.Errorf("cannot use both AppArmour profile and SELinux options")
-	}
-
-	aaProfile = strings.TrimPrefix(aaProfile, appArmorLocalhostPrefix)
-	t.g.SetProcessApparmorProfile(aaProfile)
-
-	if err := setupSELinux(&t.g, selinuxOptions); err != nil {
-		return err
-	}
-
 	for _, capb := range security.GetCapabilities().GetDropCapabilities() {
-		t.g.DropProcessCapabilityEffective(capb)
-		t.g.DropProcessCapabilityAmbient(capb)
-		t.g.DropProcessCapabilityBounding(capb)
-		t.g.DropProcessCapabilityInheritable(capb)
-		t.g.DropProcessCapabilityPermitted(capb)
+		if err := t.g.DropProcessCapabilityEffective(capb); err != nil {
+			return fmt.Errorf("could not drop capability: %v", err)
+		}
+		if err := t.g.DropProcessCapabilityAmbient(capb); err != nil {
+			return fmt.Errorf("could not drop capability: %v", err)
+		}
+		if err := t.g.DropProcessCapabilityBounding(capb); err != nil {
+			return fmt.Errorf("could not drop capability: %v", err)
+		}
+		if err := t.g.DropProcessCapabilityInheritable(capb); err != nil {
+			return fmt.Errorf("could not drop capability: %v", err)
+		}
+		if err := t.g.DropProcessCapabilityPermitted(capb); err != nil {
+			return fmt.Errorf("could not drop capability: %v", err)
+		}
 	}
 	for _, capb := range security.GetCapabilities().GetAddCapabilities() {
-		t.g.AddProcessCapabilityEffective(capb)
-		t.g.AddProcessCapabilityAmbient(capb)
-		t.g.AddProcessCapabilityBounding(capb)
-		t.g.AddProcessCapabilityInheritable(capb)
-		t.g.AddProcessCapabilityPermitted(capb)
+		if err := t.g.AddProcessCapabilityEffective(capb); err != nil {
+			return fmt.Errorf("could not add capability: %v", err)
+		}
+		if err := t.g.AddProcessCapabilityAmbient(capb); err != nil {
+			return fmt.Errorf("could not add capability: %v", err)
+		}
+		if err := t.g.AddProcessCapabilityBounding(capb); err != nil {
+			return fmt.Errorf("could not add capability: %v", err)
+		}
+		if err := t.g.AddProcessCapabilityInheritable(capb); err != nil {
+			return fmt.Errorf("could not add capability: %v", err)
+		}
+		if err := t.g.AddProcessCapabilityPermitted(capb); err != nil {
+			return fmt.Errorf("could not add capability: %v", err)
+		}
+	}
+	if t.g.Config.Linux == nil {
+		t.g.Config.Linux = new(specs.Linux)
+	}
+	t.g.Config.Linux.Seccomp = seccomp.DefaultProfile(t.g.Config) // reload seccomp profile after capabilities setup
```

Comment:
Reload means seccomp filters are loaded two times ?

Reply:
yes, but they are not loaded from anywhere, rather generated in source code. Default seccomp profile is create with `generator.New`, but because it depends on capabilities, it may change after they are set. I was wondering whether we should generate OCI config from the scratch but it appears that default config it useful in most cases, so now as a first approach I decided to simply fill seccomp twice